pkgs/by-name/pa/payloadsallthethings/package.nix

```diff
@@ -2,32 +2,66 @@
   lib,
   fetchFromGitHub,
   stdenvNoCC,
+  python3Packages,
+  yq,
+  ...
 }:
-
+let
+  inherit (python3Packages) mkdocs mkdocs-material pymdown-extensions;
+in
 stdenvNoCC.mkDerivation {
   pname = "payloadsallthethings";
-  version = "3.0-unstable-2024-01-21";
+  version = "2025.1";
 
   src = fetchFromGitHub {
     owner = "swisskyrepo";
     repo = "PayloadsAllTheThings";
-    rev = "97cfeee270395a838802fa1fcb8a4d5ffc6d6b48";
-    hash = "sha256-LRS60v0o5nPSLfGFH6P0Y5roN8Mk5/KyRF4SWTv/7Hw=";
+    tag = "4.2";
+    hash = "sha256-LBPlGfmIyzgRhUdAJmPxjDB7D8iRHcSA8Tf5teMnFzA=";
   };
 
+  patches = [ ./mkdocs.patch ];
+
+  nativeBuildInputs = [
+    mkdocs
+    mkdocs-material
+    pymdown-extensions
+
+    yq
+  ];
+
+  outputs = [
+    "out"
+    "doc"
+  ];
+
+  buildPhase = ''
+    yq -yi '.docs_dir = "source"' mkdocs.yml
+    mkdir overrides
+    mv mkdocs.yml ..
+    cd ..
+    mkdocs build --theme material
+  '';
```

Comment:
I think we can simplify this, see maybe https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/nl/nlopt/package.nix#L126-L131 ?